srk/src/polyhedronLatticeTiling.ml

```diff
@@ -0,0 +1,1486 @@
+          DD.project dimensions_to_eliminate dd
+          |> DD.enum_constraints
+          |> DD.of_constraints_closed ~man target_dim
+        in
+        match hull_alg with
+        | `GomoryChvatal ->
+           P.dd_of ~man (max_dim_in_p + 1) p
+           |> DD.integer_hull
+           |> project_and_rebuild
+        | `Normaliz ->
+           P.integer_hull `Normaliz p
+           |> P.dd_of ~man (max_dim_in_p + 1)
+           |> project_and_rebuild
+      in
+      LocalAbstraction.{ abstract; translate = restrict target_dim }
+    in
+    let (cubify, _) = Plt.cubify srk terms symbols in
+    cubify
+    |> LocalAbstraction.compose CloseStrictIneq.round_assuming_all_ints
+    |> LocalAbstraction.compose project
+
+  let by_polyreccone_and_lplh ?(man=Polka.manager_alloc_loose()) ~epsilon srk terms symbols =
+    let (cubify, _) = Plt.cubify srk terms symbols in
+    let max_dim_in_target = Array.length terms - 1 in
+    let compose = LocalAbstraction.compose in
+    cubify
+    |> compose (PltConvexHull.by_polyreccone_and_lplh ~man ~epsilon ~max_dim_in_target)
+
+  let by_lplh ?(man=Polka.manager_alloc_loose()) ~epsilon srk terms symbols =
+    let (cubify, _) = Plt.cubify srk terms symbols in
+    let max_dim_in_target = Array.length terms - 1 in
+    let compose = LocalAbstraction.compose in
+    cubify
+    |> compose (PltConvexHull.local_project_local_hull ~man ~epsilon ~max_dim_in_target)
+
+  let by_polyreccone ?(man=Polka.manager_alloc_loose()) srk terms symbols =
+    let (cubify, _) = Plt.cubify srk terms symbols in
+    let max_dim_in_target = Array.length terms - 1 in
+    let compose = LocalAbstraction.compose in
+    cubify
+    |> compose (PltConvexHull.local_project_polyreccone ~man ~max_dim_in_target)
+
+end
+
+module LocalGlobal: sig
+
+  val lift:
+    man:DD.closed Apron.Manager.t ->
+    ('a formula, 'a Interpretation.interpretation, DD.closed DD.t, int -> QQ.t) LocalAbstraction.t ->
+    'a Abstract.Solver.t -> ?bottom: DD.closed DD.t -> ('a arith_term) Array.t -> DD.closed DD.t
+
+end = struct
+
+  let print_model srk terms interp =
+    let result =
+      Array.init (Array.length terms)
+        (fun i ->
+          (terms.(i), Interpretation.evaluate_term interp terms.(i)))
+    in
+    logf ~level:`debug "model: @[%a@]@;"
+      (Format.pp_print_list
+         ~pp_sep:(fun fmt () -> Format.fprintf fmt "; ")
+         (fun fmt (t, value) ->
+           Format.fprintf fmt "(%a, %a)"
+             (Syntax.ArithTerm.pp srk) t
+             QQ.pp value))
+      (Array.to_list result)
+
+  let lift ~man local_abs solver ?bottom terms =
+    let target_dim = Array.length terms in
+    let bottom =
+      match bottom with
+      | None -> P.dd_of ~man target_dim P.bottom
+      | Some bot -> bot
+    in
+    let top = P.dd_of ~man target_dim P.top in
+    let srk = Abstract.Solver.get_context solver in
+    let phi = Abstract.Solver.get_formula solver in
+    let counter = ref 0 in
+    let models = ref [] in
```

Comment:
Why save these?

Reply:
Do you mean models? This was legacy; removed.